manifests/init.pp

```diff
@@ -12,18 +12,31 @@
 #     /mnt/mount:
 #       src: "nfs-server.local:/export/path"
 #       fstype: "nfs4"
-#       opts: "defaults,nosuid,ro"
+#       opts: "defaults,nosuid,nodev,ro"
 # ```
 #
+# @param required_pkgs
+#   Packages that need to be installed for NFS mounts to work.
+# 
 # @example
 #   include profile_nfs_client
 class profile_nfs_client (
+  Array[ String ] $masked_services,
+  Hash            $mountmap,
   Array[ String ] $required_pkgs,
-  Hash $mountmap,
 ) {
 
   ensure_packages( $required_pkgs )
 
+  each($masked_services) | $service |
+  {
+    exec { "mask_${service}":
+      command => "systemctl mask --now ${service}",
+      unless  => "systemctl status ${service} | grep Loaded | grep masked",
```

Comment:
I suspect there is a better command for doing this check.

Reply:
Is this any better?
```suggestion
      unless  => "systemctl  list-unit-files | grep masked | grep ${unit}",
```

Reply:
Maybe check out 'systemctl is-enabled"?

Reply:
Nice, yes that seems to do the trick. Thanks.